provider/common/bootstrap_test.go

```diff
@@ -398,3 +398,53 @@ func (s *BootstrapSuite) TestWaitSSHRefreshAddresses(c *gc.C) {
 		"Waiting for address\n"+
 			"(.|\n)*(Attempting to connect to 0.1.2.4:22\n)+(.|\n)*")
 }
+
+type FormatHardwareSuite struct{}
+
+var _ = gc.Suite(&FormatHardwareSuite{})
+
+func (s *FormatHardwareSuite) check(c *gc.C, hw *instance.HardwareCharacteristics, expected string) {
+	c.Check(common.FormatHardware(hw), gc.Equals, expected)
+}
+
+func (s *FormatHardwareSuite) TestNil(c *gc.C) {
+	s.check(c, nil, "")
+	s.check(c, &instance.HardwareCharacteristics{}, "")
```

Comment:
You could split this out into TestZero. The various zero tests below are all the same as this case anyway.